katello/katello/hostname-change.rb

```diff
@@ -345,12 +346,19 @@ def run
       STDOUT.puts "updating hostname in foreman installer scenarios"
       self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' /etc/foreman-installer/scenarios.d/*.yaml")
 
-      STDOUT.puts "removing last_scenario.yml file"
-      self.run_cmd("rm -rf /etc/foreman-installer/scenarios.d/last_scenario.yaml")
+      if File.exist?(last_scenario_yaml)
+        temp_dir = Dir.mktmpdir
```

Comment:
Why a temp dir rather than a temp file? Another thing to be careful about is that the answers file contains passwords and other secrets. It should not become world readable.

Reply:
Can use a tempfile & the permissions look good using that method